### ISSUE TITLE
setup:upgrade should return failure when app:config:import failed

### ISSUE THREAD

**Issue by stkrelax** [magento/magento2#23222]
### Preconditions (*)
Magento 2.3.1 CE

1. Setup app:config:dump is used
2. No Config Changes for this testcase because easy to reproduce

### Steps to reproduce (*)
1. Add an Exception to Magento\Deploy\Console\Command\App\ConfigImport\Processor
after Line 114
```
            if (!$importers || !$this->changeDetector->hasChanges()) {
              Throw new \Exception('Test'); // edited

```

2. run bin/magento setup:upgrade

### Expected result (*)
1. Cli Command returns \Magento\Framework\Console\Cli::RETURN_FAILURE
Import failed: Test
 exit code 1 

### Actual result (*)
1. Cli Command returns \Magento\Framework\Console\Cli::RETURN_SUCCESS
Import failed: Test
 exit code 0 

### Use Case
1. we are using these cli command for deploy. If there is an Error Deployment should STOP. But without errorcode we cannot stop.

### Possible Solution
Change 
```
            if ($this->deploymentConfig->isAvailable()) {
                $importConfigCommand = $this->getApplication()->find(ConfigImportCommand::COMMAND_NAME);
                $arrayInput = new ArrayInput([]);
                $arrayInput->setInteractive($input->isInteractive());
                $importConfigCommand->run($arrayInput, $output);
            }
```
To 
```
            if ($this->deploymentConfig->isAvailable()) {
                $importConfigCommand = $this->getApplication()->find(ConfigImportCommand::COMMAND_NAME);
                $arrayInput = new ArrayInput([]);
                $arrayInput->setInteractive($input->isInteractive());
                $result = $importConfigCommand->run($arrayInput, $output);
                if ($result === \Magento\Framework\Console\Cli::RETURN_FAILURE) {
                    Throw new \Magento\Framework\Exception\RuntimeException(ConfigImportCommand::COMMAND_NAME . ' failed. See previous Output.');
                }
            }


```
or use the Processor directly

**Comment by magento-engcom-team** [comment]
@ProcessEight Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [x] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @ProcessEight 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17571` were created

**Issue Available**: @ProcessEight, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @stkrelax. Thank you for your report. 
The issue has been fixed in magento/magento2#23310 by @ProcessEight in 2.3-develop branch
Related commit(s):
 - [8c4667a5ad17b8ab9c82fdc49a9c35b36ab167bf](https://github.com/magento/magento2/commit/8c4667a5ad17b8ab9c82fdc49a9c35b36ab167bf)
 - [671876ebaf4e61fbc4efdad9b1d9d67677ccc43d](https://github.com/magento/magento2/commit/671876ebaf4e61fbc4efdad9b1d9d67677ccc43d)

The fix will be available with the upcoming 2.3.3 release.
